pkg/web/apps/apps_service.go

```diff
@@ -109,3 +108,162 @@ func (a *appsService) BindingAppByApp(appId, name string) error {
 	// if is deleted so just reactive the existent app
 	return a.repository.UnDeleteAppByAppID(app.AppID)
 }
+
+func (a *appsService) checkDeviceWithThisVersion(sdkInfo *models.Device, hasVersion bool, hasApp bool) (bool, *models.Device, error) {
+	// Has no need to do the queries if has not an app or version in the DB with the SDK info
+	if !hasApp || !hasVersion {
+		return false, nil, nil
+	}
+
+	device, err := a.repository.GetDeviceByVersionAndAppID(sdkInfo.Version, sdkInfo.AppID)
+	if err != nil && err != models.ErrNotFound {
+		return false, nil, nil
+	}
+
+	return true, device, nil
+}
+
+func (a *appsService) checkVersionForThisApp(sdkInfo *models.Device, hasApp bool) (bool, *models.Version, error) {
+	// Has no need to do the queries if has not an app in the DB with the SDK info
+	if !hasApp {
+		return false, nil, nil
+	}
+
+	version, err := a.repository.GetVersionByAppIDAndVersion(sdkInfo.AppID, sdkInfo.Version)
+	if err != nil && err != models.ErrNotFound {
+		return false, nil, nil
+	}
+	return true, version, nil
+}
+
+func (a *appsService) checkAnyVersionOfAppForThisDevice(sdkInfo *models.Device, hasVersion bool, hasApp bool) (bool, *models.Device, error) {
+	// Has no need to do the queries if has not an app or version in the DB with the SDK info
+	if !hasApp || hasVersion {
+		return false, nil, nil
+	}
+
+	device, err := a.repository.GetDeviceByDeviceIDAndAppID(sdkInfo.DeviceID, sdkInfo.AppID)
+	if err != nil && err != models.ErrNotFound {
+		return false, nil, nil
+	}
+	return true, device, nil
+}
+
+func (a *appsService) checkIfHasApp(sdkInfo *models.Device) (bool, *models.App, error) {
+	app, err := a.repository.GetAppByAppID(sdkInfo.AppID)
+
+	if err != nil && err != models.ErrNotFound {
+		return false, nil, nil
+	}
+	return true, app, nil
+}
+
+// Init call made from the SDK
+// Response: Version details
+// If the app, version and or device are not tracked in the database then it will be made
+func (a *appsService) InitClientApp(sdkInfo *models.Device) (*models.Version, error) {
+	hasApp, app, err := a.checkIfHasApp(sdkInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	//If the app be deleted then is required re-active it
+	if hasApp && app.DeletedAt != "" {
+		a.repository.UnDeleteAppByAppID(app.AppID)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	hasVersion, version, err := a.checkVersionForThisApp(sdkInfo, hasApp)
+	if err != nil {
+		return nil, err
+	}
+
+	//Check if the version is not disable before continue
+	// If it is disable than just response it
```

Comment:
@davidffrench and @craicoverflow 

To attend the last comment of the cris in the trello card where he is saying that we should increment the values when the version is disabled is required just remove this check then shows that all will as defined.